assets/src/edit-story/components/form/advancedDropDown/list/list.js

```diff
@@ -125,19 +125,23 @@ function OptionList({
     () =>
       new window.IntersectionObserver(
         (entries) => {
-          entries.forEach((entry) => {
-            if (entry.isIntersecting) {
-              setUserSeenOptions(addUniqueEntry(entry.target.dataset.option));
-            }
-          });
+          if (onObserve) {
+            const newlySeenOptions = entries
+              .filter((entry) => entry.isIntersecting)
+              .map((entry) => entry.target.dataset.option);
+            userSeenOptions.current = addUniqueEntries(...newlySeenOptions)(
```

Comment:
Maybe change the function so that it could be just used `addUniqueEntries(...newlySeenOptions, userSeenOptions.current)`?

Reply:
Yeah, I will do that, the curried version played nice with the useState, but now it's just used in this one place.